templates/client-daemonset.yaml

```diff
@@ -235,6 +277,33 @@ spec:
           - name: aclconfig
             mountPath: /consul/aclconfig
       {{- end }}
+      {{- if .Values.global.tls.enabled }}
+      - name: client-tls-init
+        image: {{ .Values.global.image }}
+        env:
+        - name: HOST_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+        command:
+          - "/bin/sh"
+          - "-ec"
+          - |
+            cd /consul/tls/client
+            consul tls cert create -client \
+              -dc={{ .Values.global.datacenter }} \
+              -domain={{ .Values.global.domain }} \
+              -additional-ipaddress=${HOST_IP} \
+              -ca=/consul/tls/ca/tls.crt \
+              -key=/consul/tls/ca/tls.key
```

Comment:
Hmm, that's an interesting question. For clients, these will be regenerated every time clients are restarted. So hopefully you restart your clients at least once in a year 🤞 

For servers, it might make sense to extend the validity period since those secrets are generated only once. We can extend those to two years.

Yeah, I could definitely rename the files :)

Reply:
> For clients, these will be regenerated every time clients are restarted. So hopefully you restart your clients at least once in a year

Nice, nvm then.